src/states/elder/mod.rs

```diff
@@ -368,6 +368,14 @@ impl Elder {
         }
     }
 
+    fn finalise_prune(&mut self) -> Result<(), RoutingError> {
+        let _ = self.chain.finalise_prune();
+        self.chain.reset_candidate();
```

Comment:
Why are we reseting candidate? does pruning impact how we accept candidates?

Reply:
You are right, it shall not reset candidate. I will correct it.